proposals/42-human-readable-mutable-names.md

```diff
@@ -0,0 +1,235 @@
+  - Given how DNS is implemented in OS and ISPs, queries are most likely sent in plain text and can be spoofed in LAN or MITMd at the ISP level. At the time of writing this proposal DNS over HTTPS is not supported, and it is not trivial for regular user to set up a custom DNS server to change this behavior.
+
+**TLDR:** **We need to fix and future-proof the way mutable names work in our stack:**
+- Make IPNS useful on its own, and inside of DNSLink records.
+    - DNS TXT record should be set only once, updates should be published over IPNS.
+- Future-proof the way we do human readable names by
+    - leveraging DNS protocol (not the DNS network run by ICANN) for interop with existing and future user agents and naming systems
+    - allowing flexible configuration to improve security (DoH) and interop (DNSLink)
+
+
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+
+- Browser is the  main distribution channel for dapps.
+- Dapps are simply a subset of all static websites loaded from IPFS.
+- Most of websites hosted on IPFS have DNSLink set up for human-readable name.
+- We have mainstream browsers like Opera and [Brave](https://brave.com/ipfs-support/) ship support for `ipfs://` and `ipns://` URIs. 
+    - Human readable names like `ipns://en.wikipedia-on-ipfs.org` work thanks to DNSLink
+    - This removes huge UX friction for dapp developers. 
+    - Vendors like Brave are committed to [surfacing DNSLink supportin the UI](https://github.com/brave/brave-browser/issues/13609) which will improve onboarding even further.
+- Alternatives to [ICANN](https://en.wikipedia.org/wiki/ICANN)  exist, and we want them to work with our stack to decrease dependency on a single organization.
+    - [ENS](https://ens.domains) provides custom TLDs (`.eth`) and leverage DNSLink for interop with IPFS and delegated lookups with clients that are unable to run own blockchain resolver.
+        - Updating `ipfs-ns` `contenthash` is a chore and costs extra (gas etc). 
+        - Using `ipns-ns` means setting `contenthash` only once, introduces cost savings and simplifies publishing. 
+    - We see multiple actors in the space providing either own TLDs or attempting to replace ICANN as the top-level authority: https://unstoppabledomains.com , http://opennic.org, https://handshake.org etc.
+    - We want to enable innovation in the decentralized naming space. This means removing ourselves as gatekeepers of what a valid domain name is.
+    - When embedded in user agent (Brave) we want to follow user choices regarding DNS resolution.
+    
+
+
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+- Improving human-readable naming removes huge DX/UX friction for dapp users and developers. 
+    - Messaging collapses to "Install Brave or Opera and open ipns://example.com"
```

Comment:
This is nice in that it doesn't rely on the ICANN network or HTTP gateways (thus working with other DNS networks and/or without HTTP gateways).  That said, the story is already good on the ICANN network right with HTTP gateways (e.g., A user can open Brave and hit ipns://libp2p.io ), right?  Basically, I want to make sure I understand this right, and it seems like the value prop here is for people who care about full decentralization.

Reply:
Supporting non-ICANN TLDs is a value prop for decentralization crowd, but even ICANN TLDs will benefit from this in privacy/security dimensions:

- virtually all default DNS people have is some plaintext DNS server from ISP
- browsers  are moving towards using custom DoH resolvers instead of the OS one to avoid privacy leaks over unencrypted DNS 
  - if go-ipfs does not do the same, we are privacy leakers: https://github.com/ipfs-shipyard/ipfs-companion/issues/900
- public gateways are running go-ipfs too
  - there is value for gateway operator in being able to change resolver to something that hides DNS queres from their cloud provider: limits privacy leak, and is way harder to spoof, as it is no longer a matter of compromising DNS at the same provider